reference/forms/types/collection.rst

```diff
@@ -268,6 +268,13 @@ delete_empty
 
 **type**: ``Boolean`` **default**: ``false``
 
+.. caution::
+
+    Delete empty will only remove items when the normalized value is null.
+    If your `type`_ is a compound form type, you need to have the :ref:`required <reference-form-option-required>`
+    option set to ``false`` inside `options`_, See :ref:`empty_data <reference-form-option-empty-data>`.
```

Comment:
In the above text, when I mean 'options' I'm referring to 'entry_options', it is because I'm making the pull request to syfmony 2.7.